Text embedding model v2 response causes deserialization error

As of https://github.com/softwaremill/sttp-openai/pull/77, the embedding model expects the response output to be one of the two models defined int the ADT. The API returns the full model string with v2 though (`text-embedding-ada-002-v2`), causing deserialization to fail. Note that the input still needs to be without v2 appended.

It might makes sense to return to a string representation for the response, or allow parsing into `CustomEmbeddingsModel`.

https://github.com/softwaremill/sttp-openai/blob/268e09174957922be23bcbff390ad549c3c585af/core/src/main/scala/sttp/openai/requests/embeddings/EmbeddingsRequestBody.scala#L33-L43

```json
{
  "object": "list",
  "data": [
    {
      "object": "embedding",
      "index": 0,
      "embedding": [
        0.030170223,
        -0.008959178,
        -0.017063277,
        -0.0039148545,
        .....
      ]
    }
  ],
  "model": "text-embedding-ada-002-v2",
  "usage": {
    "prompt_tokens": 2,
    "total_tokens": 2
  }
}
```

How to reproduce:

```scala
//> using lib "com.softwaremill.sttp.openai::core:0.0.6"

import sttp.client4._
import sttp.openai.OpenAISyncClient
import sttp.openai.requests.embeddings.EmbeddingsRequestBody.*

object Embeddings extends App {
  // Create an instance of OpenAISyncClient providing your API secret-key
  val openAI: OpenAISyncClient = OpenAISyncClient("<api-key>")

  val embeddingsBody = EmbeddingsBody(
    EmbeddingsModel.TextEmbeddingAda002,
    EmbeddingsInput.SingleInput("Hello world")
  )

  val embeddingsResponse = openAI.createEmbeddings(embeddingsBody)

  println(embeddingsResponse)
}
```

Error:

```
Caused by: sttp.openai.OpenAIExceptions$OpenAIException$DeserializationOpenAIException: Could not deserialize: text-embedding-ada-002-v2
        at sttp.openai.OpenAIExceptions$OpenAIException$DeserializationOpenAIException$.apply(OpenAIExceptions.scala:22)
        at sttp.openai.OpenAIExceptions$OpenAIException$DeserializationOpenAIException$.apply(OpenAIExceptions.scala:24)
        at sttp.openai.json.SttpUpickleApiExtension$.deserializeJsonSnake$$anonfun$1(SnakePickle.scala:65)
```

--- a/core/src/main/scala/sttp/openai/requests/embeddings/EmbeddingsRequestBody.scala
+++ b/core/src/main/scala/sttp/openai/requests/embeddings/EmbeddingsRequestBody.scala
@@ -30,7 +30,7 @@ object EmbeddingsRequestBody {
         jsonValue =>
           SnakePickle.read[ujson.Value](jsonValue) match {
             case Str(value) =>
-              byEmbeddingsModelValue.getOrElse(value, throw DeserializationOpenAIException(new Exception(s"Could not deserialize: $value")))
+              byEmbeddingsModelValue.getOrElse(value, CustomEmbeddingsModel(value))
             case e => throw DeserializationOpenAIException(new Exception(s"Could not deserialize: $e"))
           }
       )

